cmd/clusterctl/client/tree/discovery.go

```diff
@@ -65,10 +67,14 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
 	// Adds control plane
-	controlPLane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
-	if err == nil {
-		tree.Add(cluster, controlPLane, ObjectMetaName("ControlPlane"), GroupingObject(true))
+	if cluster.Spec.ControlPlaneRef == nil {
+		return nil, errors.New(fmt.Sprintf("spec.controlPlaneRef for Cluster %v is nil, this is unexpected", client.ObjectKeyFromObject(cluster)))
```

Comment:
```suggestion
		return nil, errors.Errorf("spec.controlPlaneRef for Cluster %v is nil, this is unexpected", client.ObjectKeyFromObject(cluster))
```
Should fix the linter warning and aligns to how we usually do it :)

Reply:
Nice, I missed the lint issue!